astroquery/xmm_newton/core.py

```diff
@@ -0,0 +1,338 @@
+            example: 0144090201
+        filename : string
+            file name to be used to store the file
+        verbose : bool
+            optional, default 'False'
+            flag to display information about the process
+        level : string
+            level to download, optional, by default everything is downloaded
+            values: ODF, PPS
+        instname : string
+            instrument name, optional, two characters, by default everything
+            values: OM, R1, R2, M1, M2, PN
+        instmode : string
+            instrument mode, optional
+            examples: Fast, FlatFieldLow, Image, PrimeFullWindow
+        filter : string
+            filter, optional
+            examples: Closed, Open, Thick, UVM2, UVW1, UVW2, V
+        expflag : string
+            exposure flag, optional, by default everything
+            values: S, U, X(not applicable)
+        expno : integer
+            exposure number with 3 digits, by default all exposures
+            examples: 001, 003
+        name : string
+            product type, optional, 6 characters, by default all product types
+            examples: 3COLIM, ATTTSR, EVENLI, SBSPEC, EXPMAP, SRCARF
+        datasubsetno : character
+            data subset number, optional, by default all
+            examples: 0, 1
+        sourceno : hex value
+            source number, optional, by default all sources
+            example: 00A, 021, 001
+        extension : string
+            file format, optional, by default all formats
+            values: ASC, ASZ, FTZ, HTM, IND, PDF, PNG
+
+
+        Returns
+        -------
+        None if not verbose. It downloads the observation indicated
+        If verbose returns the filename
+        """
+
+        link = self.data_aio_url + "obsno=" + observation_id
+
+        link = link + "".join("&{0}={1}".format(key, val)
+                              for key, val in kwargs.items())
+
+        if verbose:
+            log.info(link)
+
+        response = None
+        try:
+            response = self._request('GET', link, save=True, cache=True)
+        except HTTPError:
+            log.error("Unable to access URL {0}".format(link))
+
+        if response:
+            if filename is None:
+                filename = observation_id + ".tar"
+
+            log.info("Copying file to {0}...".format(filename))
+            path = os.getcwd()
+            destfile = os.path.join(path)
+            file = None
+            try:
+                os.makedirs(destfile, exist_ok=True)
+                file = open(os.path.join(destfile, filename), "w+")
+            except OSError:
+                log.error("Creation of the directory %s failed" % destfile)
+            else:
+                if verbose:
+                    log.info("Successfully created the directory %s " %
+                             destfile)
+                shutil.copy(response,
+                            os.path.join(path, filename))
```

Comment:
The whole block here could be shortened to:
```suggestion
            path = os.getcwd()
            shutil.copy(response, os.path.join(path, filename))
```
or, even shorter and simpler,
```
            shutil.copy(response, os.path.join('.', filename))
```